src/mass.rs

```diff
@@ -252,6 +256,42 @@ impl Measurement for Mass {
     }
 }
 
+#[cfg(feature = "from_str")]
+impl FromStr for Mass {
+    type Err = std::num::ParseFloatError;
+
+    /// Create a new Mass from a string
+    /// Plain numbers in string are considered to be Kilograms
+    fn from_str(val: &str) -> Result<Self, Self::Err> {
+        if val.is_empty() {
+            return Ok(Mass::from_kilograms(0.0));
+        }
+
+        let re = Regex::new(r"(?i)\s*([0-9.]*)\s?([a-zμ]{1,3})\s*$").unwrap();
+        if let Some(caps) = re.captures(val) {
+            let float_val = caps.get(1).unwrap().as_str();
+            return Ok(
+                match caps.get(2).unwrap().as_str().to_lowercase().as_str() {
+                    "ug" | "μg" => Mass::from_micrograms(float_val.parse::<f64>()?),
+                    "mg" => Mass::from_milligrams(float_val.parse::<f64>()?),
+                    "ct" => Mass::from_carats(float_val.parse::<f64>()?),
+                    "g" => Mass::from_grams(float_val.parse::<f64>()?),
+                    "kg" => Mass::from_kilograms(float_val.parse::<f64>()?),
+                    "t" => Mass::from_metric_tons(float_val.parse::<f64>()?),
+                    "gr" => Mass::from_grains(float_val.parse::<f64>()?),
```

Comment:
Hmm, I am a bit unsure about this one. People may use gr. for grams too.

Reply:
You are correct. Wikipedia says "The SI does not support the use of abbreviations such as "gr" (which is the symbol for grains)"

Reply:
Oh. That's for grains. I think it's correct.

Reply:
More info https://en.wikipedia.org/wiki/Gram

Reply:
Ok!